sklearn/utils/__init__.py

```diff
@@ -639,3 +639,35 @@ def is_scalar_nan(x):
     # Redondant np.floating is needed because numbers can't match np.float32
     # in python 2.
     return bool(isinstance(x, (numbers.Real, np.floating)) and np.isnan(x))
+
+
+# Use at least float64 for the accumulating functions to avoid precision issues;
+# see https://github.com/numpy/numpy/issues/9393
+# The float64 is also retained as it is in case the float overflows
+def safe_acc_op(op, x, *args, **kwargs):
```

Comment:
Please make this private, maybe more verbose (`_safe_accumulate_op`) and move it to `utils.extmath`.